backend/routes/generations.py

```
@@ -335,6 +378,22 @@ async def stream_speech(
     tts_model = get_tts_backend_for_engine(engine)
     model_size = data.model_size or "1.7B"
 
+    text = data.text
+    if data.personality and getattr(profile, "personality", None):
+        try:
+            llm_result = await personality.rewrite_as_profile(
+                profile.personality,
+                data.text,
+            )
+        except ValueError as e:
+            raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)